### ISSUE TITLE
Included tests make the package large

### ISSUE THREAD

**Issue by ryanvolz** [conda-forge/pillow-feedstock#117]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Recent builds of the `pillow` package, which include the tests and test images, weigh in at ~45 MB. Without the tests included, the package occupies less than 1 MB of space. The tests don't get installed into the environment, so the main effect is a bigger download and more space occupied in the package cache. Since nobody is likely to want to run the tests themselves from the packaged tarball, I think that shrinking the package back down to its previous size would be a big benefit (particularly since `pillow` is a dependency of `matplotlib-base`, which I assume many people will install).

Conda-build has a `--no-copy-test-source-files` which would allow the tests to still run in the feedstock and prevent them from ending up in the package, but I don't know if there is a way to use that. Another possible solution could be to create a separate output (e.g. `pillow-tests`) and run the tests through that. That way they would still be available in a package, but the majority of users would not have to use the extra space.

### Installed packages

```shell
N/A
```

### Environment info

```shell
N/A
```

**Comment by h-vetinari** [comment]
Unless there's a simple way to pass arguments to `conda-build` from cf infrastructure (CC @conda-forge/core), I think the `pillow-tests` approach would be the most reasonable.

Thanks for the report!

**Comment by hmaarrfk** [comment]
Yeah, i've been trying to think of a good name of this extra package.

For the conda-package, I gave up on finding a solution:
https://github.com/conda-forge/conda-feedstock/blob/main/recipe/meta.yaml#L5

The tests were never run prior to adding that flag.

**Comment by hmaarrfk** [comment]
my issue wit hthe `tests` package, is that it may overlap with a real package name from upstream.

I think maybe `pillow__tests__` ? or something like that. Fewer chances of collisions.

**Comment by h-vetinari** [comment]
> my issue with the `tests` package, is that it may overlap with a real package name from upstream.

TBH, I don't think that's an issue. And if upstream ever were to publish something like that, we could build some deprecation for our output. The scipy-feedstock would [similarly](https://github.com/conda-forge/scipy-feedstock/issues/160) take `scipy-tests`.

**Comment by hmaarrfk** [comment]
Ok. The tests name sounds fine

**Comment by hmaarrfk** [comment]
I guess we are at the point where a PR would be accepted.

**Comment by h-vetinari** [comment]
Down from around 40MB, that's a pretty crazy improvement! 🤯
